src/context.cpp

```diff
@@ -596,61 +699,172 @@ namespace librealsense
             if ( profiles.size() > 0 )
             {
                 _dev->open( realdds_profiles );
+
             }
 
             software_sensor::open( profiles );
         }
 
+        void handle_video_data( realdds::topics::device::image && dds_frame,
+                                std::shared_ptr< stream_profile_interface > prof, std::string name )
+        {
+            rs2_software_video_frame rs2_frame = {};
+
+            // prof parameter holds the real data, rs2_software_video_frame forces us to hold a pointer to it.
+            // Because we use syncer, not calling on_video_frame in the lifetime of this function, we need a shared_ptr
+            // that the syncer will hold till using the frame.
+            std::shared_ptr< rs2_stream_profile > prof_holder = std::make_shared< rs2_stream_profile >();
+            prof_holder->profile = prof.get();
+            rs2_frame.profile = prof_holder.get();
+
+            //Copying from dds into LibRS space, same as copy from USB backend.
+            //TODO - use memory pool or some other frame allocator
+            rs2_frame.pixels = new uint8_t[dds_frame.size];
+            if( !rs2_frame.pixels )
+                throw std::runtime_error( "Could not allocate memory for new frame" );
+            memcpy( rs2_frame.pixels, dds_frame.raw_data.data(), dds_frame.size );
+            // The way to release allocated memory
+            rs2_frame.deleter = []( void * ptr ) { delete[] ptr; };
+            // Info sent with image topic
+            rs2_frame.stride = dds_frame.height > 0 ? dds_frame.size / dds_frame.height : dds_frame.size;
+            rs2_frame.bpp = dds_frame.width > 0 ? rs2_frame.stride / dds_frame.width : rs2_frame.stride;
+            rs2_frame.frame_number = std::stoi( dds_frame.frame_id );
+
+            if( _md_enabled )
+                _stream_name_to_video_syncer[name].enqueue_frame( std::move( rs2_frame ), prof_holder );
+            else
+                on_video_frame( rs2_frame );
+        }
+
+        void add_video_frame_metadata( rs2_software_video_frame & frame, const realdds::topics::flexible_msg & dds_md )
+        {
+            // Always expected metadata
+            frame.timestamp = rsutils::json::get< rs2_time_t >( dds_md.json_data(), "timestamp" );
```

Comment:
`json_data()` actually parses the binary data every time, so I suggest calling it once and then querying it

Reply:
OK, I thought that repeated calls to unchanged object are optimized, but I will not count on it and keep it explicitly.